components/permissions/permission_expiration_key.cc

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/permissions/permission_expiration_key.h"
+
+#include <utility>
+
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_util.h"
+
+namespace permissions {
+
+namespace {
+
+base::Time ParseExpirationTime(const std::string& key_str) {
+  int64_t expiration_time = 0;
+  if (!base::StringToInt64(key_str, &expiration_time)) {
+    return base::Time();
+  }
+  return base::Time::FromDeltaSinceWindowsEpoch(
+      base::TimeDelta::FromMicroseconds(expiration_time));
+}
+
+std::string ExpirationTimeToStr(base::Time expiration_time) {
+  return std::to_string(
+      expiration_time.ToDeltaSinceWindowsEpoch().InMicroseconds());
+}
+
+}  // namespace
+
+PermissionExpirationKey::PermissionExpirationKey(base::Time time)
+    : time_(time) {}
+PermissionExpirationKey::PermissionExpirationKey(std::string domain)
```

Comment:
`const&`

Reply:
disagree. this is the proper case for taking by value, the value should and will be moved at runtime.